src/opnsense/mvc/tests/app/models/OPNsense/Base/Constraints/UniqueConstraintTest.php

```diff
@@ -0,0 +1,175 @@
+        $constraint = new UniqueConstraint();
+        $this->addChildNode('UniqueTest', $container);
+        $addFields = [];
+        $fields_added = false;
+        foreach ($nodes as $name => $value) {
+            $node = new TextField(null, $name);
+            $node->setRequired($this->valuesRequired ? "Y" : "N");
+            if ($name === array_key_first($nodes)) {
+                $constraint->setOption('node', $node);
+                $constraint->setOption('name', $name);
+                $constraint->setOption('ValidationMessage', 'Validation Failed');
+            } else {
+                $addFields[] = $name;
+                $fields_added = true;
+            }
+            $node->setValue($value);
+            $container->addChildNode($name, $node);
+        }
+
+        if ($fields_added) {
+            $constraint->setOption('addFields', $addFields);
+        }
+
+        $this->uniqueConstraints[] = $constraint;
+    }
+
+    public function setRequired($required)
+    {
+        $this->valuesRequired = $required;
+    }
+
+    public function validate()
+    {
+        $validator = new \OPNsense\Base\Validation();
+        foreach ($this->uniqueConstraints as $idx => $constraint) {
+            $validator->add($idx, $constraint);
+        }
+        $msgs = $validator->validate([]);
+
+        return $msgs;
+    }
+}
+
+class UniqueConstraintTest extends \PHPUnit\Framework\Testcase
+{
+    public function testNonEqualAndEqualValues()
+    {
+        $container = new UniqueTestContainer();
+        $container->addNode(['unique_test' => 'value1']);
+        $container->addNode(['unique_test' => 'value2']);
+
+        $msgs = $container->validate();
+        $this->assertEquals(0, $msgs->count());
+
+        $container->addNode(['unique_test' => 'value1']);
+
+        $msgs = $container->validate();
+        $this->assertEquals(1, $msgs->count());
+    }
+
+    public function testMultipleNonEqualAndEqualValues()
+    {
+        $container = new UniqueTestContainer();
+        $container->addNode(['unique_test' => 'value1', 'unique_test2' => 'value2']);
+        $container->addNode(['unique_test' => 'value1', 'unique_test2' => 'value3']);
+
+        $msgs = $container->validate();
+        $this->assertEquals(0, $msgs->count());
+
+        $container->addNode(['unique_test' => 'value1', 'unique_test2' => 'value2']);
+
+        $msgs = $container->validate();
+        $this->assertEquals(1, $msgs->count());
+    }
+
+    public function testEmptyValuesNotRequiredAndRequired()
+    {
+        $container = new UniqueTestContainer();
+        $container->addNode(['unique_test' => '']);
+        $container->addNode(['unique_test' => '']);
+
+        $msgs = $container->validate();
+
+        $this->assertEquals(0, $msgs->count());
+
+        $container = new UniqueTestContainer();
+        $container->setRequired(true);
+        $container->addNode(['unique_test' => '']);
+        $container->addNode(['unique_test' => '']);
```

Comment:
```suggestion
        $container->setRequired(true);
```

shouldn't this work? :)

Reply:
if we change the logic of UniqueTestContainer, yes

Reply:
only if it's no trouble, don't need to spend more time than necessary

Reply:
https://github.com/opnsense/core/pull/6875/commits/4e05bcfc7506b24ca2489e14facf46ca361fd174 ;)